schema/samcli.json

```diff
@@ -247,9 +279,96 @@
+                  "type": "string",
+                  "description": "AWS SAM template which references built artifacts for resources in the template. (if applicable)",
+                  "default": "template.[yaml|yml|json]"
+                },
+                "output_template_file": {
+                  "title": "output_template_file",
+                  "type": "string",
+                  "description": "The path to the file where the command writes the output AWS CloudFormation template. If you don't specify a path, the command writes the template to the standard output."
+                },
+                "s3_bucket": {
+                  "title": "s3_bucket",
+                  "type": "string",
+                  "description": "AWS S3 bucket where artifacts referenced in the template are uploaded."
+                },
+                "image_repository": {
+                  "title": "image_repository",
+                  "type": "string",
+                  "description": "AWS ECR repository URI where artifacts referenced in the template are uploaded."
+                },
+                "image_repositories": {
+                  "title": "image_repositories",
+                  "type": "string",
+                  "description": "Mapping of Function Logical ID to AWS ECR Repository URI.\n\nExample: Function_Logical_ID=ECR_Repo_Uri\nThis option can be specified multiple times."
+                },
+                "s3_prefix": {
+                  "title": "s3_prefix",
+                  "type": "string",
+                  "description": "Prefix name that is added to the artifact's name when it is uploaded to the AWS S3 bucket."
+                },
+                "kms_key_id": {
+                  "title": "kms_key_id",
+                  "type": "string",
+                  "description": "The ID of an AWS KMS key that is used to encrypt artifacts that are at rest in the AWS S3 bucket."
+                },
+                "use_json": {
+                  "title": "use_json",
+                  "type": "boolean",
+                  "description": "Indicates whether to use JSON as the format for the output AWS CloudFormation template. YAML is used by default."
+                },
+                "force_upload": {
+                  "title": "force_upload",
+                  "type": "boolean",
+                  "description": "Indicates whether to override existing files in the S3 bucket. Specify this flag to upload artifacts even if they match existing artifacts in the S3 bucket."
+                },
+                "resolve_s3": {
+                  "title": "resolve_s3",
+                  "type": "boolean",
+                  "description": "Automatically resolve AWS S3 bucket for non-guided deployments. Enabling this option will also create a managed default AWS S3 bucket for you. If one does not provide a --s3-bucket value, the managed bucket will be used. Do not use --guided with this option."
```

Comment:
> Do not use --guided with this option.

I'm guessing this comes from the deploy command (or rather the option was written with the deploy command in mind and we are reusing it here)

Reply:
Yup, that seems to be the case. It uses a decorator method to produce the `resolve_s3` click option, and the help text seems specifically catered to deploy. We could change that text if need be, to be more specific about "If running with deploy, do not use --guided", but I don't think it's completely necessary.